src/System.ComponentModel.EventBasedAsync/tests/AsyncOperationFinalizerTests.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading;
+using Xunit;
+
+namespace System.ComponentModel.Tests
+{
+    public class AsyncOperationFinalizerTests : RemoteExecutorTestBase
+    {
+        [Fact]
+        public void Finalizer_GetViaReflection_IsPresentForCompatability()
+        {
+            MethodInfo finalizer = typeof(AsyncOperation).GetMethod("Finalize", BindingFlags.NonPublic | BindingFlags.Instance);
+            Assert.NotNull(finalizer);
+        }
```

Comment:
Correct me if I'm wrong, the tests do exactly that. They use the following sync context that the finalizer does indeed call back to when we manually invoke it. This test you commented on is just a sanity checker.

```cs
public class OperationCompletedTracker : SynchronizationContext
{
    public int OperationCompletedCounter { get; set; }

    public override void OperationCompleted() => OperationCompletedCounter++;
}
```

In terms of guaranteeing that the finalizer is invoked without reflection, I don't know what the best way to do that is